openedx/openedx-platform
# Fixes TNL-3664 by collapsing actions dropdown after an action item is clicked.

### Fixes [TNL-3644](https://openedx.atlassian.net/browse/TNL-3644): Course settings dropdown doesn't disappear if you close modals
#### Components affected:
- LMS
#### Steps to verify:
1. Go to your student dashboard page in the LMS
2. Click the "gear" icon and select "Unenroll"
3. On the Unenroll modal, exit out of the modal

Note that the settings menu is now collapsed. It stayed open before the fix.
#### Additional discussion

Normally, when we fix a bug, we would like to add a test alongside the fix, to be sure that the bug remains fixed. You will note that I do not include a test with this PR. My rationale is that the code in [legacy.js](https://github.com/edx/edx-platform/blob/69dc51859e57e458e1a64717224343ff90e19298/lms/static/js/dashboard/legacy.js) is not designed to be testable, so I do not see how to test the new behaviour without rewriting major part of [legacy.js](https://github.com/edx/edx-platform/blob/69dc51859e57e458e1a64717224343ff90e19298/lms/static/js/dashboard/legacy.js), [_dashboard_course_listing.html](https://github.com/edx/edx-platform/blob/69dc51859e57e458e1a64717224343ff90e19298/lms/templates/dashboard/_dashboard_course_listing.html), and perhaps introducing a new template file. My concern is that such change introduces much more risk (read: potential bugs and problems) than it actually fixes. Especially when it is done by someone who is completely new to the project such as me.

In comparison, the conservative bugfix is trivial. So I propose it here.
#### Legal

This PR is covered by OpenCraft contributor agreement.

@antoviaque
